src/utils/hooks/useKubevirtDataPod/utils/utils.ts

```diff
@@ -0,0 +1,23 @@
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { WatchK8sResource } from '@openshift-console/dynamic-plugin-sdk';
+
+import { PROXY_KUBEVIRT_URL } from './constants';
+
+const rsMapper = {};
+
+export const registerResourceVersion = (key: string, rs: string) => {
+  rsMapper[key] = rs > rsMapper[key] ? rsMapper[key] : rs;
+  return;
+};
+export const getResourceVersion = (key: string) => rsMapper[key];
+
+export const constructURL = (watchOptions: WatchK8sResource, query?: string) => {
+  const { groupVersionKind, name, namespace } = watchOptions || {};
+  const baseUrl = `${PROXY_KUBEVIRT_URL}apis/${groupVersionKind?.group}/${groupVersionKind?.version}/`;
+  const namespaceUrl = `${namespace ? `namespaces/${namespace}/` : ''}`;
+  const nameUrl = `${!name ? 's' : `/${name}`}`;
```

Comment:
why `s`?

Reply:
url of /virtualmachines will give all
but if you want a specific machine, meaning have a name, the URL is changed to /virtualmachine/name

so "s" is for "get me all" I dont use names